ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java

```diff
@@ -144,8 +146,12 @@ private static boolean isTransformedTouchPointInView(
       localX = localXY[0];
       localY = localXY[1];
     }
-    if ((localX >= 0 && localX < (child.getRight() - child.getLeft()))
-        && (localY >= 0 && localY < (child.getBottom() - child.getTop()))) {
+    Rect hitSlopRect = new Rect();
+    if (child instanceof ReactViewGroup && ((ReactViewGroup) child).getHitSlopRect() != null) {
```

Comment:
I think it's better to be explicit here and have one comparison for views that implement HitSlopRect and those that don't. That way, we don't have the additional Rect allocation every single time we check if a touch is inside a View.